pkg/virt-handler/isolation/process.go

```diff
@@ -0,0 +1,51 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package isolation
+
+import "github.com/mitchellh/go-ps"
+
+// filterProcessByPPID returns chiled processes of the given PID's.
+func filterProcessByPPID(processes []ps.Process, pids []int) []ps.Process {
+	filterProcessParentPid := map[int]struct{}{}
+	for _, pid := range pids {
+		filterProcessParentPid[pid] = struct{}{}
+	}
+
+	var filteredProcesses []ps.Process
+	for _, process := range processes {
+		if _, ok := filterProcessParentPid[process.PPid()]; ok {
+			filteredProcesses = append(filteredProcesses, process)
+		}
+	}
+
+	return filteredProcesses
+}
+
+// filterProcessByPPID returns processes that runs the given executable.
+func filterProcessByExecutable(processes []ps.Process, exectutable string) []ps.Process {
```

Comment:
Looking at the usage, it is enough for this function to return the first detected process with the mentioned executable.
Generalizing it now just complicates things unnecessarily.

And in such a case, this can be named: `lookupProcessByExecutable`.